### ISSUE TITLE
As University of Trento I would like that the non linear calibration formula + calibration values of the air quality sensors on the A22 are updated

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/bdp-elaborations#32]
Following elaboration should be improved: https://github.com/noi-techpark/bdp-elaborations/tree/main/environment-a22-non-linear-calibration

according to new calibration values + formula, as provided in this attachment

[240402_UNITN_Airqino_coefficients_winter_2023-2024.xlsx](https://github.com/noi-techpark/bdp-elaborations/files/15016968/240402_UNITN_Airqino_coefficients_winter_2023-2024.xlsx)

We should delete all "historical" elaborations, i.e. consider the installations that have been done recently (see attachment
[240326_PosizioneSensori.xlsx](https://github.com/noi-techpark/bdp-elaborations/files/15061553/240326_PosizioneSensori.xlsx). E.g. AQ12 - delete all elaborations after 11.03.2024

**Comment by clezag** [comment]
@rcavaliere I've just updated the parameter map in testing (it's not running yet for some other reason). From what I understood, it's just like #27, only that now all stations use the new formula for the NO2 calculations.

Regarding deleting the data, I assume we just delete the data type NO2_alphasense and leave the rest, right?

**Comment by rcavaliere** [comment]
@clezag exactly! Only those with period = 3600

**Comment by rcavaliere** [comment]
@clezag thanks for your work! Yes, for these 14 stations the calibration curves for the other pollutants still need to be calculated, so please implement the "ignore" logic.

**Comment by clezag** [comment]
@rcavaliere done in testing now, seems good to me

**Comment by rcavaliere** [comment]
@clezag wonderful, I had a look, the values are absolutely reasonable. I made some comparison with reference averages made available by the reference air quality stations and are perfectly in line. Let's go in production!

**Comment by clezag** [comment]
@rcavaliere up in production

**Comment by clezag** [comment]
@rcavaliere done in testing

**Comment by rcavaliere** [comment]
Very good, let's go in production!

**Comment by rcavaliere** [comment]
@clezag sorry I didn't tag you here... do you want to do it today or let's wait for Tuesday next week?

**Comment by clezag** [comment]
@rcavaliere will do it next week

**Comment by clezag** [comment]
@rcavaliere moved into production

I've noticed that all the stations I checked where stuck on the 14.05.2024 and didn't get any new data (at the raw data mqtt side). We've had an internal issue around that time that affected most data collectors, and unfortunately this data collector seems to be written in a way where it wasn't able to recover, but also didn't report any errors or crash.

I've now restarted it and it seems to be receiving data again, so it will take a moment for the elaboration pipeline to catch up.

**Comment by rcavaliere** [comment]
OK, I leave the issue open in case we get some other comments from the BrennerLEC partners, internally we can consider the issue closed for the moment

**Comment by rcavaliere** [comment]
@clezag the calibration of the sensor AQ12  ("Stazione_KM012-200 (AUGEG4_AIRQ12)" seems to be wrong, the elaboration is nearly always zero. Can you check if you applied it correctly and let me know?

**Comment by clezag** [comment]
@rcavaliere I've checked the parameters themselves, and did a few test calculation by hand, in my opinion the elaboration does what it's supposed to do.

For the next update, would it be possible to have them submit a pull request [directly on the csv](https://github.com/noi-techpark/bdp-elaborations/blob/main/environment-a22-non-linear-calibration/src/processorParameters.csv)? Or if they are not technically inclined, at least mail us the full updated csv. There is a lot of margin for error when copying stuff together from various excel files with different number formats etc.

**Comment by rcavaliere** [comment]
@clezag did this PR went into production? https://github.com/noi-techpark/bdp-elaborations/actions/runs/9357205755
The sensor is still returning incorrect values, let me know if we still need to fix something here.

Another request: can you reactivate please the function that provides our elaboration back to the MQTT server of A22?

**Comment by clezag** [comment]
@rcavaliere sorry I did not see this commit.
I've merged it into production and re-elaborated the data.
Now it looks plausible

**Comment by rcavaliere** [comment]
Perfect! Now we just miss to reactivate the service towards the MQTT server and we can close this ticket

**Comment by rcavaliere** [comment]
@clezag, yeah you are right, it is not so immediate. But can't you activate the connection to the MQTT broker from the elaboration module?

**Comment by clezag** [comment]
@rcavaliere I'm afraid no. The elaboration has been rewritten from the ground up (in python vs the original Java), so the MQTT part would have to be implemented from scratch. In addition, the elaboration runs once every hour and works on hourly aggregated data only. The data collector worked a 60 second period on raw data. So a good part of the logic would have to be changed.

Just updating the formula+parameters of the data collector and switching the MQTT back on will be easier, it's just that from now on, someone has to remember and update both applications every time something changes

**Comment by rcavaliere** [comment]
@clezag ok, let's discuss this together. We could think to ask our counterpart to foresee something different at the MQTT broker side, in order to receive hourly elaborations...

**Comment by rcavaliere** [comment]
@clezag I think that in relation to the topic on how to best provide the elaborations outputs to the MQTT server the best way to proceed would be to implement this logic at the level of the elaboration module. In this way we completely decouple the data collection and data elaboration tasks. I.e., when elaborations are computed they are sent back to the MQTT server. 
For the other side this approach would be OK, in the sense that we could completely differentiate the mechanisms of data reading / writing with the MQTT server

**Comment by rcavaliere** [comment]
@clezag can you test the interface with the MQTT server using the new IP addresses you shared me? A22 should have given us grant to connect. Please test and let me know

**Comment by clezag** [comment]
@rcavaliere It's still not connecting from kubernetes. I've checked from both test and production

**Comment by clezag** [comment]
@rcavaliere It's fixed now, and I'm receiving data. Thanks for your work!

**Comment by clezag** [comment]
Waiting for https://github.com/noi-techpark/bdp-core/issues/287
